### PR TITLE
Update Key Management docs for Swift

### PR DESCRIPTION
**Pull request by acekyd**
This pull request primarily focuses on expanding the language support in the key management documentation and adding a new test suite for Swift. The most significant changes include adding Swift as a language option in the `LanguageSwitcher` component, including Swift code snippets in the key management documentation, and introducing a new Swift test suite for key management.

Addition of Swift support:

* [`site/docs/web5/build/decentralized-identifiers/key-management.mdx`](diffhunk://#diff-83dafc02d54448c17f0398fd3c40c396dca039e0cf578a27d052091a2330a988L7-R7): Added Swift as an option in the `LanguageSwitcher` component.
* [`site/docs/web5/build/decentralized-identifiers/key-management.mdx`](diffhunk://#diff-83dafc02d54448c17f0398fd3c40c396dca039e0cf578a27d052091a2330a988L58-R76): Included Swift code snippets in the key management documentation.

New Swift test suite:

* [`site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift`](diffhunk://#diff-ac4debf6560f691c97d6358dbe065cdd4afb982cb45fa47ff534a3dff6a6e762R1-R36): Introduced a new Swift test suite for key management. This test suite includes a test for initializing key management with an `InMemoryKeyManager` and a `BearerDID`.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206944752955378

### CODE REVIEW

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import XCTest
+@testable import DevSiteTestSuite
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:2]: i dont think this is needed. can you double check?

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+final class KeyManagementTests: XCTestCase {
+    func testInitializeKeyManagement() throws {
+        let keyManager = InMemoryKeyManager()
+        let bearerDID = try DIDJWK.create(keyManager: keyManager)
+        let testPortableDID = try bearerDID.export()
+
+        // :snippet-start: initializeKeyManagementSwift
+        func initializeKeyManagement(portableDID: PortableDID?) throws -> BearerDID {
+            //Creates a new instance of the InMemoryKeyManager
+            let keyManager = InMemoryKeyManager() // if production, use a different key manager
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:15]: by "different" what do you mean? We should give clear guidance

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import XCTest
+@testable import DevSiteTestSuite
+import Mocker
+import Web5
+
+final class KeyManagementTests: XCTestCase {
+    func testInitializeKeyManagement() throws {
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:7]: any reason it's not called `initKeyManagement` like the JS and KT examples?

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        let keyManager = InMemoryKeyManager()
+        let bearerDID = try DIDJWK.create(keyManager: keyManager)
+        let testPortableDID = try bearerDID.export()
+
+        // :snippet-start: initializeKeyManagementSwift
+        func initializeKeyManagement(portableDID: PortableDID?) throws -> BearerDID {
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:13]: any reason it's not called `initKeyManagement` like the JS and KT examples?

Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:13]: should it take an `env` variable like the other examples where we show to use a different key manager for Production?

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift (hunk 1 above)
Comment (acekyd) [reply site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:2]: Yes testable isn't needed indeed.

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift (hunk 4 above)
Comment (acekyd) [reply site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:13]: The hold up with this was trying to figure out what example to show for the production keyManager. On it.

**Review by angiejones — CHANGES_REQUESTED** [review]
Install instructions missing

image

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        let keyManager = InMemoryKeyManager()
+        let bearerDID = try DIDJWK.create(keyManager: keyManager)
+        let testPortableDID = try bearerDID.export()
+
+        // :snippet-start: initializeKeyManagementSwift
+        func initKeyManagement(env: String?, portableDID: PortableDID?) throws -> BearerDID {
+            let keyManager = InMemoryKeyManager()
+
+            if env == "production" {
+                // if production, use a more secure and production-ready key manager.
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:17]: @kirahsapong @frankhinek do we have (or plan to have) production-ready key manager support for the web5-swift SDK? Seems iOS would have something, yeah?

Comment (angiejones) [reply site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:17]: Ace, looks like Adam implemented this

https://github.com/TBD54566975/web5-swift/blob/1353f59e105ef09b4705f8675ce329f0f7e4c104/Sources/Web5/Crypto/KeyManager/Local/KeychainKeyManager.swift#L5

Comment (acekyd) [reply site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:17]: Thanks for this. Updated the code and all looks good. Looking forward to any additional security insights.

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    func testInitializeKeyManagement() throws {
+        let keyManager = InMemoryKeyManager()
+        let bearerDID = try DIDJWK.create(keyManager: keyManager)
+        let testPortableDID = try bearerDID.export()
+
+        // :snippet-start: initializeKeyManagementSwift
+        func initKeyManagement(env: String?, portableDID: PortableDID?) throws -> BearerDID {
+            let keyManager: LocalKeyManager
+
+            if env == "production" {
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:16]:
```suggestion
            // Determine which key manager to use based on the environment
            if env == "production" {
```

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                // if production, use a more secure and production-ready key manager.
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:17]:
```suggestion

```

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                keyManager = KeychainKeyManager()
+            }
+            else {
+                keyManager = InMemoryKeyManager()
+            }
+
+            if portableDID != nil {
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:24]:
```suggestion
            // Initialize or load a DID
            if portableDID != nil {
```

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                return try DIDJWK.import(
+                    keyManager: keyManager,
+                    portableDID: portableDID!
+                )
+
+            }
+            else {
+                return try DIDJWK.create(keyManager: keyManager)
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:33]:
```suggestion
                // Create a new DID
                return try DIDJWK.create(keyManager: keyManager)
```

File: site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+            if env == "production" {
+                // if production, use a more secure and production-ready key manager.
+                keyManager = KeychainKeyManager()
+            }
+            else {
+                keyManager = InMemoryKeyManager()
+            }
+
+            if portableDID != nil {
+                return try DIDJWK.import(
```
Comment (angiejones) [inline site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/KeyManagementTests.swift:26]:
```suggestion
                 // Load existing DID
                return try DIDJWK.import(
```

**Review by angiejones — APPROVED** [review]

**Pull request merged**